### ISSUE TITLE
'NoneType' object has no attribute 'LGBM_BoosterCreate' issue

### ISSUE THREAD

**Issue by aparnack** [Microsoft/LightGBM#1305]
Please search your question on previous issues, [stackoverflow](https://stackoverflow.com/questions/tagged/lightgbm) or other search engines before you open a new one.

For bugs and unexpected issues, please provide following information, so that we could reproduce on our system.

## Environment info
Operating System: Windows 10
CPU: core i7, 8th Gen
C++/Python/R version: mingw-64 installed on April 6th 2018,Python 3.6

## Error Message:
'NoneType' object has no attribute 'LGBM_BoosterCreate'

## Reproducible examples
lgbm=lightgbm.train({'num_leaves':150, 'objective':'binary','max_depth':7},train_lgb)

## Steps to reproduce

1. Prepare train set
2. Prepare params dictionary
3. Call training command lightgbm.train()

**Comment by guolinke** [comment]
it seems you didn't install lightgbm correctly, since the `_LIB` is None.
https://github.com/Microsoft/LightGBM/blob/ebb07f019a611ec9d3712aadbc9e725faaa9c78c/python-package/lightgbm/basic.py#L1302

**Comment by aparnack** [comment]
Hi
I deleted the entire thing and reinstalled as per https://lightgbm.readthedocs.io/en/latest/Installation-Guide.html  - this is what I had followed earlier as well. 
It still gives the same error.
xgboost works, and installation process is same. Hence I dont understand what went wrong in this installation. There are hardly any discussions in stackoverflow on lightgbm issues

Thanks
Aparna

**Comment by guolinke** [comment]
I think you are using python-package with mingw64 in windows 10? right? 
if yes, you should install from source, with option --mingw, refer to https://github.com/Microsoft/LightGBM/tree/master/python-package#install-from-github
```
git clone --recursive https://github.com/Microsoft/LightGBM.git
cd LightGBM/python-package
python setup.py install --mingw
```

Or you can install vc runtime, and install from pip.

**Comment by aparnack** [comment]
Hi Guolin
Even this new method doesnt work.
I had first tried the commands in git bash last time, that gives sh.exe error. But in the second try it ran (this was mentioned in the lightgbm documentation). So this time I did everything in dos command (cmd.exe).  The cmd installation gave this error. I think there's some serious problem with my whole system. [My anaconda itself doesnt get invoked.] So, I will get back to this later. Let us not spend time on this, if someone happens to see this issue and suggest, great!

Traceback (most recent call last) : 
File "setup.py", line 282, in <module> 
' Topic 
Scientific/ Engineering 
. Artificial Intelligence' ] ) 
File "C: line 129, 
in setup 
return distutils . core. 
File "C: line 148, 
di st . run_commands ( ) 
File "C: .py", line 955, 
self. run_command ( cmd) 
File "C: .py", line 974, 
cmd_obj . run( ) 
File "setup.py", line 2e9, in run 
install . run( self) 
in setup 
in run commands 
in run command 
File "C: .py", line 65, 
orig. install . run( self) 
File "C: .py", line 557, in run 
self. run_command ( cmd_name ) 
File "C: line 313, in run_command 
self. di st ribution. run_command ( command) 
File "C: .py", line 974, in run_command 
cmd_obj . run( ) 
in run 
File "C: line 12, 
outfiles = 
self. install ( ) 
File "setup.py", line 164, in install 
src = 
IndexError: list index out of range 
in run
